students/chris_delapena/lesson03/mailroom.py

```diff
@@ -0,0 +1,122 @@
+If the user types a name not in the list, add that name to the data structure
+and use it.
+If the user types a name in the list, use it.
+Once a name has been selected, prompt for a donation amount.
+Convert the amount into a number; it is OK at this point for the program to
+crash if someone types a bogus amount.
+Add that amount to the donation history of the selected user.
+Finally, use string formatting to compose an email thanking the donor for
+their generous donation. Print the email to the terminal and return to the
+original prompt.
+
+If the user (you) selected “Create a Report,” print a list of your donors,
+sorted by total historical donation amount.
+Include Donor Name, total donated, number of donations, and average donation
+amount as values in each row. You do not need to print out all of each donor’s
+donations, just the summary info.
+Using string formatting, format the output rows as nicely as possible. The
+end result should be tabular (values in each column should align with those
+above and below).
+After printing this report, return to the original prompt.
+At any point, the user should be able to quit their current task and return
+to the original prompt.
+From the original prompt, the user should be able to quit the script cleanly.
+"""
+
+import sys
+
+donor_db = [("William Gates, III", [653772.32, 12.17]),
+            ("Jeff Bezos", [877.33]),
+            ("Paul Allen", [663.23, 43.87, 1.32]),
+            ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0])
+            ]
+
+prompt = "\n".join(("Mail Room Program version 1.0",
+            "Please choose from below options:",
+            "1 - Send a Thank you",
+            "2 - Create a Report",
+            "3 - Quit",
+            ">>> "))
+
+prompt_name = "\n".join(("Please enter the full name of the donor",
+                "or type 'list' to see a full list of donor names",
+                "or type 'q' to quit."
+                ">>> "))
+
+prompt_amount = "\n".join(("What was the donation amount?",
+                ">>> "))
+
+def thank_you():
+    donor_name = input(prompt_name)
+    if donor_name.upper() == 'Q':
+        return
+    elif donor_name.upper() == "LIST":
+        for donor in donor_db:
+            print('-'*60)
+            print(donor[0])
+            print('-'*60)
+    else:
+        for i in donor_db:
+            if donor_name == i[0]: #Use i[0] to handle tuple
+                donation_amount = input(prompt_amount)
+                print('-'*60)
+                print(f"Dear {donor_name}, thank you for your generous donation of ${donation_amount}."
+                " Regards, the Club Owners")
+                print('-'*60)
+                return
+        else:
+            print('-'*60)
+            print(f"Adding new donor {donor_name}")
+            print('-'*60)
+            new_donor = (donor_name, [])
+            donor_db.append(new_donor)
+            return
```

Comment:
you are missing a feature here - you should ask for donation in both case (new or existing donor)